### PR TITLE
[Source Gen] Test Cleanup

### PR DESCRIPTION
**Pull request by satvu**
### Issue describing the changes in this PR

Not changing the tests, just cleaning up the strings with C# 11 string literals ([see here](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-11.0/raw-string-literal)).

### Pull request checklist

* [x] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **should not** be added to the release notes for the next release
  * [ ] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] I have added all required tests (Unit tests, E2E tests)

### Additional information

Additional PR information

### CODE REVIEW

File: sdk/Sdk.Generators/FunctionMetadataProviderGenerator.Emitter.cs [hunk 1]
```diff
@@ -49,7 +49,7 @@ public string Emit(IReadOnlyList<GeneratorFunctionMetadata> funcMetadata, Cancel
                     AddRegistrationExtension(indentedTextWriter);
 
                     indentedTextWriter.Indent--;
-                    indentedTextWriter.WriteLine("}");
+                    indentedTextWriter.Write("}");
```
Comment (satvu) [inline sdk/Sdk.Generators/FunctionMetadataProviderGenerator.Emitter.cs:52]: Removes the white space at the end of the generated file

**Comment by fabiocav** [comment]
I've added my changes for the generator, getting rid of the `IndentedTextWriter` there. I'll wait for your changes to land first, as I'll need to update the tests for small tweaks I made there.

File: test/Sdk.Generator.Tests/FunctionMetadataProviderGeneratorTests/IntegratedTriggersAndBindingsTests.cs [hunk 2]
```diff
@@ -252,41 +256,43 @@ public static IHostBuilder ConfigureGeneratedFunctionMetadataProvider(this IHost
...
+                        public string Name { get; }
+                    }
                 }
-            }".Replace("'", "\"");
+                """;
 
                 string expectedGeneratedFileName = $"GeneratedFunctionMetadataProvider.g.cs";
-                string expectedOutput = @"// <auto-generated/>
+                string expectedOutput = """
+// <auto-generated/>
```
Comment (fabiocav) [inline test/Sdk.Generator.Tests/FunctionMetadataProviderGeneratorTests/IntegratedTriggersAndBindingsTests.cs:295]: Align this?

File: test/Sdk.Generator.Tests/FunctionMetadataProviderGeneratorTests/StorageBindingTests.cs [hunk 3]
```diff
@@ -139,27 +141,29 @@ namespace FunctionApp
...
                         }
                     }
-                }".Replace("'", "\"");
+                }
+                """;
 
                 string expectedGeneratedFileName = $"GeneratedFunctionMetadataProvider.g.cs";
-                string expectedOutput = @"// <auto-generated/>
+                string expectedOutput = """
+// <auto-generated/>
```
Comment (fabiocav) [inline test/Sdk.Generator.Tests/FunctionMetadataProviderGeneratorTests/StorageBindingTests.cs:166]: Align?

**Review by fabiocav — APPROVED** [review]

**Pull request merged**